programs.wake

```diff
@@ -0,0 +1,119 @@
+package freedom_e_sdk
+
+from wake import _
+
+export tuple FreedomESDKProgram =
+  export Name: String
+  export Kind: String
+  export Sources: List Path
+
+export def getFreedomESDKProgramSourceDir program =
+  def name = program.getFreedomESDKProgramName
+  def kind = program.getFreedomESDKProgramKind
+  "{kind}-{name}"
+
+export def makeFreedomESDKProgram name kind =
+  def program = FreedomESDKProgram name kind Nil
+  def sourceDir = program.getFreedomESDKProgramSourceDir
+  def programSources = sources sourceDir `.*`
```

Comment:
This makes the assumption that the source directories are at the root of the wit/wake workspace, which may be problematic in the future, like I alluded to above.